src/test/java/com/wikia/webdriver/testcases/Anonymization/AnonymizationTests.java

```diff
@@ -0,0 +1,192 @@
+    UserRegistration.registerUserEmailConfirmed(user);
+
+    SpecialAnonymizationUserPage anonymizationStaff = new SpecialAnonymizationUserPage().open();
+    anonymizationStaff.loginAs(User.STAFF);
+    anonymizationStaff.fillFutureAnon(qanon)
+        .submitAnonymization();
+
+    Assertion.assertStringContains(anonymizationStaff.getAnonConfirmation(), qanon);
+
+    UserProfilePage userProfilePage = new UserProfilePage().open(qanon);
+    Assertion.assertTrue(userProfilePage.getUserNotExists().equals(String.format("User account \"%s\" does not exist or has never logged in on this wiki.", qanon)));
+  }
+
+  @Test
+  public void anonymizationWikiEdits() {
+
+    Credentials credentials = new Credentials();
+    String timestamp = Long.toString(DateTime.now().getMillis());
+    String qanon = "QAanon" + timestamp;
+    String passw = "makak";
+    SignUpUser
+        user = new SignUpUser(qanon, credentials.emailAnonymousUserTestWikia, passw,
+                              LocalDate.of(1990, 3, 19));
+    UserRegistration.registerUserEmailConfirmed(user);
+
+    new WikiBasePageObject().loginAs(user.getUsername(), user.getPassword(), wikiURL);
+
+    new ArticleContent(user.getUsername()).push("Test" + timestamp, "AnonymizationTest");
+
+    ArticleHistoryPage articleHistoryPage = new ArticleHistoryPage().open("AnonymizationTest");
+    String id = articleHistoryPage.getHistoryID(user.getUsername());
+    Assertion.assertTrue(articleHistoryPage.isUserInHistory(user.getUsername()));
+
+    articleHistoryPage.logOut();
+
+    SpecialAnonymizationUserPage anonymizationStaff = new SpecialAnonymizationUserPage().open();
+    anonymizationStaff.loginAs(User.SUS_STAFF);
+    anonymizationStaff.fillFutureAnon(qanon)
+        .submitAnonymization();
+
+    Assertion.assertStringContains(anonymizationStaff.getAnonConfirmation(), qanon);
+
+    articleHistoryPage = new ArticleHistoryPage().open("AnonymizationTest");
+    Assertion.assertFalse(articleHistoryPage.isUserInHistory(user.getUsername()));
+  }
+
+
+  @Test
+  public void anonyizationMessageWallRemoved() {
+
+    Credentials credentials = new Credentials();
+    String timestamp = Long.toString(DateTime.now().getMillis());
+    String qanon = "QAanon" + timestamp;
+    String passw = "makak";
+    SignUpUser
+        user = new SignUpUser(qanon, credentials.emailAnonymousUserTestWikia, passw,
+                              LocalDate.of(1990, 3, 19));
+    UserRegistration.registerUserEmailConfirmed(user);
+
+    new WikiBasePageObject().loginAs(user.getUsername(), user.getPassword(), wikiURL);
+
+    MessageWall wall = new MessageWall().open(user.getUsername());
+    MiniEditorComponentObject mini = wall.triggerMessageArea(true);
+    String message = String.format("%s%s", PageContent.MESSAGE_WALL_MESSAGE_PREFIX, timestamp);
+    String title = String.format("%s%s", PageContent.MESSAGE_WALL_TITLE_PREFIX, timestamp);
+    mini.switchAndWrite(message);
+    wall.setTitle(title);
```

Comment:
If `setTitle() method would return MessageWall (and `this`) you could chain this methods like: `wall.setTitle(title).submit()`. It's really more readable